src/MarkdownTextInput.web.tsx

```diff
@@ -1,22 +1,438 @@
-      ref={ref}
-    />
-  );
+const getCurrentCursorPosition = CursorUtils.getCurrentCursorPosition;
+
+const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
+  (
+    {
+      accessibilityLabel,
+      accessibilityLabelledBy,
+      accessibilityRole,
+      autoCapitalize = 'sentences',
+      autoCorrect = true,
+      blurOnSubmit = false,
+      clearTextOnFocus,
+      dir = 'auto',
+      multiline = false,
+      markdownStyle,
+      onBlur,
+      onChange,
+      onChangeText,
+      onClick,
+      onFocus,
+      onKeyPress,
+      onSelectionChange,
+      onSubmitEditing,
+      placeholder = '\u200B', // Zero-width space to prevent div from collapsing
+      placeholderTextColor = '#ccc',
+      selectTextOnFocus,
+      spellCheck,
+      style = {},
+      value,
+    },
+    ref,
+  ) => {
+    const divRef = useRef<HTMLDivElement | null>(null);
+    const currentlyFocusedField = useRef<HTMLDivElement | null>(null);
+    const contentSelection = useRef<Selection | null>(null);
+    const className = `react-native-live-markdown-input-${multiline ? 'multiline' : 'single-line'}`;
+    const history = useRef(new MarkdownInputHistory(20));
+    const flattenStyle = useMemo(() => StyleSheet.flatten(style), [style]);
+    const parseText = useCallback(
+      (target: HTMLDivElement, text: string | null, customMarkdownStyles: MarkdownStyle, cursorPosition: number | null = null, shouldAddToHistory = true) => {
+        if (text === null) {
+          return {text: target.innerText, cursorPosition: null};
+        }
+        const parsedText = ParseUtils.parseText(target, text, cursorPosition, customMarkdownStyles, !multiline);
+        if (history.current && shouldAddToHistory) {
+          history.current.debouncedAdd(parsedText.text, parsedText.cursorPosition);
+        }
+        return parsedText;
+      },
+      [multiline],
+    );
+
+    const processedMarkdownStyle = React.useMemo(() => {
+      const newMarkdownStyle = StyleUtils.mergeMarkdownStyleWithDefault(markdownStyle, true);
+      if (divRef.current) {
+        parseText(divRef.current, divRef.current.innerText, newMarkdownStyle);
+      }
+      return newMarkdownStyle;
+    }, [markdownStyle]);
+
+    const undo = useCallback(
+      (target: HTMLDivElement) => {
+        const item = history.current.undo();
+        return parseText(target, item ? item.text : null, processedMarkdownStyle, item ? item.cursorPosition : null, false).text;
+      },
+      [processedMarkdownStyle],
+    );
+
+    const redo = useCallback(
+      (target: HTMLDivElement) => {
+        const item = history.current.redo();
+        return parseText(target, item ? item.text : null, processedMarkdownStyle, item ? item.cursorPosition : null, false).text;
+      },
+      [processedMarkdownStyle],
+    );
+
+    // We have to process value porperty since contentEditable div adds one additional '\n' at the end of the text if we are entering new line
```

Comment:
```suggestion
    // We have to process value property since contentEditable div adds one additional '\n' at the end of the text if we are entering new line
```